lib/mayaUsd/utils/utilDictionary.cpp

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "utilDictionary.h"
+
+namespace MAYAUSD_NS_DEF {
+
+namespace DictUtils {
+
+using namespace PXR_NS;
+
+/// Extracts a bool at \p key from \p userArgs, or false if it can't extract.
+bool extractBoolean(const VtDictionary& userArgs, const TfToken& key)
```

Comment:
All this code was moved as-is from jobArgs.cpp, except from renaming them with a extract prefix instead of an underscore and putting them in a namespace.